### ISSUE TITLE
Will klaus become a "git web manager"?

### ISSUE THREAD

**Issue by fpemud** [jonashaag/klaus#255]
I'm looking for a web based personal git repository manager.
Pure personal usage only.
I don't want to set up a GitLab server, it's has too much redundant functions such as issues, merge requests, wiki, email...

Currently klaus is a "git web viewer", on the same level as gitweb/cgit.
Will it evolve to a "git web manager" in future? which means the following functions will be added:
(1) create/import/delete/rename repository
(2) edit online (#206, response from @jonashaag seems positive)

It seems there's no product is on this "git web manager" direction.

**Comment by jonashaag** [comment]
Maybe in 20 years :-) Nobody is investing a lot of time in the project in the past few years.

**Comment by jelmer** [comment]
Once you start down this path, it's hard to not end up with some of those other functions..

allowing editing online would pretty much make klaus function as a wiki...
in order to allow editing and repository management, you'd need authentication and authorization. Once you allow people who don't have write access to a repository to make edits, you're also adding merge requests.

**Comment by fpemud** [comment]
gitlab is more of a "team collaborate platform". The only difference between github is that it can be self-hosted.
What I'm looking for is a "web based git repo manage tool" and I think many people would need it. There can be a clear border between these 2 kinds of products.

wiki is not a part of git tree, so I don't think wiki function is in the scope.
Authentication and authorization function is ok, but as a simple tool, there should be no "allow people who don't have write access to a repository to make edits" scenario.

**Comment by cpressey** [comment]
Just wanted to note this:

Klaus currently bills itself as "a simple, easy-to-set-up Git web viewer that Just Works™" and the first bullet point of its feature list is "Super easy to set up -- no configuration required".

I'm not sure it's possible to add authentication to a product and also to continue to make claims like those.

If people like Klaus because of these goals, then more sophisticated features might appear in forks of Klaus rather than in the Klaus project itself.

**Comment by jonashaag** [comment]
I agree with @fpemud that there might be a need for a simple, local-only, single-user repo management tool to find, create, clone, modify, push, etc. Git repos. I guess lots of Git users essentially have a folder structure on their local machine like this:

```
~/dev/
  - repo1
  - repo2
  - largeproject1
    - repo1
    - repo2
  - ...
```

And then they use `cd && ls`, or VSCode, or whatever, to manage these. Which works okay but isn't Git/repo centric. You could use some repo centric like GitLab, Gitea, etc. but it's not very well integrated into a local machine workflow and it also doesn't Just Work.

That said, unless someone else is willing to invest a lot of time into evolving Klaus in this direction, it's not going to happen. As with most open source projects, I started Klaus to satisfy my personal needs, which is essentially a nice way to look at Git history and diffs on a local machine or small private server. Klaus does this pretty well, so I personally have little interest in adding lots of things at this point. I'm happy to discuss plans with anyone who is interested though!
